src/Angular.js

```diff
@@ -886,16 +886,7 @@ function copy(source, destination) {
     if (isArray(source)) {
       destination = [];
       needsRecurse = true;
-    } else if (isTypedArray(source)) {
-      destination = new source.constructor(source);
-    } else if (isDate(source)) {
-      destination = new Date(source.getTime());
-    } else if (isRegExp(source)) {
-      destination = new RegExp(source.source, source.toString().match(/[^\/]*$/)[0]);
-      destination.lastIndex = source.lastIndex;
-    } else if (isFunction(source.cloneNode)) {
-        destination = source.cloneNode(true);
-    } else {
+    } else if (!(destination = copyType(source))) {
```

Comment:
I think this will not work as intended with `new Boolean(false)` nor with `new String('')`